src/pybind/mgr/stats/fs/perf_stats.py

```diff
@@ -166,6 +173,52 @@ def notify(self, cmdtag):
             self.log.debug("client_metadata={0}, to_purge={1}".format(
                 self.client_metadata['metadata'], self.client_metadata['to_purge']))
 
+    def notify_fsmap(self):
+        #Reregister the user queries when there is a new rank0 mds
+        gid_state = FSPerfStats.get_rank0_mds_gid_state(self.module.get('fs_map'))
+        if not gid_state:
+            return
+        rank0_gid, state = gid_state
+        if (rank0_gid and (rank0_gid != self.prev_rank0_gid)):
+            if (state != 'up:active'):
+                return
+            #ronk0 MDS is up:active
+            ua_last_updated = datetime.now()
+            if (self.rqtimer and self.rqtimer.is_alive()):
+                self.rqtimer.cancel()
+            self.rqtimer = Timer(REREGISTER_TIMER_INTERVAL,
+                                 self.re_register_queries, args=(rank0_gid, ua_last_updated,))
+            self.rqtimer.start()
+
+    def re_register_queries(self, rank0_gid, ua_last_updated):
+        #reregister queries if the metrics are the latest. Otherwise reschedule the timer and
+        #wait for the empty metrics
+        if self.mx_last_updated >= ua_last_updated:
+            self.log.debug("reregistering queries...")
+            for filter_spec in list(self.user_queries.keys()):
+                with self.lock:
```

Comment:
should this cover the entire check (i.e., before checking `self.mx_last_updated >= ua_last_updated`)?